remotes/docker/resolver.go

```diff
@@ -222,19 +222,11 @@ func (r *countingReader) Read(p []byte) (int, error) {
 var _ remotes.Resolver = &dockerResolver{}
 
 func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocispec.Descriptor, error) {
-	refspec, err := reference.Parse(ref)
-	if err != nil {
-		return "", ocispec.Descriptor{}, err
-	}
-
-	if refspec.Object == "" {
```

Comment:
For remote.Fetcher/Pusher, it can be done with digest. But for `Resolve` API, it still needs to check `Object`.